### PR TITLE
BACKLOG-21810 : refactor sitemap generation

### PR DESCRIPTION
**Pull request by dgriffon**
- use static JCR Sessions
 - store sitemap as a binary (to use the Datastore instead of a DB blob)
 - improve sitemap entries generation
 - 
https://jira.jahia.org/browse/BACKLOG-21810

### CODE REVIEW

File: src/main/java/org/jahia/modules/sitemap/utils/Utils.java [hunk 1]
```diff
@@ -119,72 +119,92 @@ public static Set<String> getSitemapRoots(RenderContext renderContext, String lo
     /**
      * @return sitemap entries that are publicly accessible
      */
-    public static Set<SitemapEntry> getSitemapEntries(RenderContext renderContext, String rootPath, Locale locale) throws RepositoryException {
-        final Set<SitemapEntry> result = new LinkedHashSet<>();
+    public static void getSitemapEntries(Map<Locale, JCRSessionWrapper> sessionPerLocale, RenderContext renderContext, String rootPath, Map<Locale, Set<SitemapEntry>> entriesByLocale, Map<String, Set<SitemapEntry>> entriesByPath) throws RepositoryException {
```
Comment (jsinovassin) [inline src/main/java/org/jahia/modules/sitemap/utils/Utils.java:122]: Is it possible to rename this method, a name like fillSitemapEntries is more meaningful for this method as it's not returning results anymore

Comment (dgriffon) [reply src/main/java/org/jahia/modules/sitemap/utils/Utils.java]: Indeed, it has been renamed. Thank you!

File: src/main/java/org/jahia/modules/sitemap/utils/Utils.java [hunk 2]
```diff
@@ -96,95 +104,98 @@ public static String encodeSitemapLink(String URIPath, boolean shouldBeDecodedFi
...
-                                    if (node != null && node.isNodeType(DEDICATED_SITEMAP_MIXIN) && !node.getPath().equals(rootPath)) {
-                                        excludedPath.add(node.getPath());
-                                    } else if (node != null && isValidEntry(node, renderContext)) {
-                                        result.add(buildSiteMapEntry(node, locale, guestUser, renderContext));
-                                    }
+
+        JCRSessionWrapper session = sessionPerLocale.get(null);
+        // Get excluded path
+        String queryFromRootPath = String.format("select * FROM [%s] as sel WHERE ISDESCENDANTNODE(sel, '%s')", DEDICATED_SITEMAP_MIXIN, StringUtils.replace(rootPath, "'", "''"));
+        QueryResult results = session.getWorkspace().getQueryManager().createQuery(queryFromRootPath, Query.JCR_SQL2).execute();
```
Comment (jsinovassin) [inline src/main/java/org/jahia/modules/sitemap/utils/Utils.java:135]: The results of this query seems to be never used

File: pom.xml [hunk 3]
```diff
@@ -1,234 +1,225 @@
...
-        <require-capability>osgi.extender;filter:="(osgi.extender=org.jahia.bundles.blueprint.extender.config)"
-        </require-capability>
-        <jahia-depends>site-settings-seo=[3.0,99],graphql-dxm-provider</jahia-depends>
-        <jahia-module-signature>MCwCFA8mIQkUfH0hHWkqsEuMA1LWaWhYAhRoNUCxJznNeUn0LeXX1vz3/FOuYg==</jahia-module-signature>
-        <yarn.arguments>build:production</yarn.arguments>
-        <export-package>org.jahia.modules.sitemap.utils</export-package>
-        <sonar.sources>src/javascript</sonar.sources>
-        <import-package>
-            graphql.annotations.annotationTypes;version="[7.2,99)",
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (jsinovassin) [inline pom.xml:2]: I think that you have an indent issue on the pom.xml file

File: src/main/java/org/jahia/modules/sitemap/services/impl/SitemapServiceImpl.java [hunk 4]
```diff
@@ -179,17 +196,36 @@ public String getSitemap(String siteKey, String key) {
...
+                cacheNode.setProperty("sitemapFile", session.getValueFactory().createBinary(stream));
+            } catch (Throwable e) {
+                throw new RuntimeException(e);
+            }
             session.save();
             return null;
         });
     }
 
+    private void removeSitemaps() {
```
Comment (jsinovassin) [inline src/main/java/org/jahia/modules/sitemap/services/impl/SitemapServiceImpl.java:214]: This function has been removed in the previous commit https://github.com/Jahia/sitemap/pull/192/files#diff-ba5fe80985eb4b164cfe8bfede2fb565d9d23c3f50c91de70101365ba3b6be1dL198.
You maybe missed this when doing the rebase

File: tests/createPages.groovy [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import java.util.*
```
Comment (jsinovassin) [inline tests/createPages.groovy:1]: Did you manage to use this file somewhere in the tests? or it's a file which can be used for manual tests?

**Review by jsinovassin — CHANGES_REQUESTED** [review]

File: src/main/java/org/jahia/modules/sitemap/utils/Utils.java (hunk 2 above)
Comment (dgriffon) [reply src/main/java/org/jahia/modules/sitemap/utils/Utils.java:135]: thank you!

File: pom.xml (hunk 3 above)
Comment (dgriffon) [reply pom.xml:2]: Indeed, updated!

File: src/main/java/org/jahia/modules/sitemap/services/impl/SitemapServiceImpl.java (hunk 4 above)
Comment (dgriffon) [reply src/main/java/org/jahia/modules/sitemap/services/impl/SitemapServiceImpl.java:214]: I missed that one while rebasing .. true

File: tests/createPages.groovy (hunk 5 above)
Comment (dgriffon) [reply tests/createPages.groovy:1]: Yes, that's for manual testing only

**Review by jsinovassin — APPROVED** [review]

**Pull request merged**